### PR TITLE
XW-516 | Move typings to JSDoc

### PR DESCRIPTION
**Pull request by tomasznapieralski**
https://wikia-inc.atlassian.net/browse/XW-516

### CODE REVIEW

File: front/scripts/baseline/mercury/utils/isPrimitive.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
 module Mercury.Utils {
+	/**
+	 * @param {any} val
+	 * @returns {boolean} isPrimitive
```
Comment (hakubo) [inline front/scripts/baseline/mercury/utils/isPrimitive.ts]: I think you don't specify a var name in `@returns` as outside the function the var name have no effect

File: front/scripts/baseline/mercury/utils/loadScript.ts [hunk 2]
```diff
@@ -2,11 +2,21 @@
 /// <reference path='../../mercury.d.ts' />
 
 module Mercury.Utils {
+	/**
+	 * @throws URIError exception
+	 * @param {ErrorEvent} error
+	 * @returns {void}
```
Comment (hakubo) [inline front/scripts/baseline/mercury/utils/loadScript.ts]: this I think should be undefined, there is on `void ` type in JS

Comment (hakubo) [inline front/scripts/baseline/mercury/utils/loadScript.ts]: also if we'd say that JSDoc are mendatory for every function
I'd say we should rethink if we want to enforce
`@returns {undefined}` 

sounds like noise

and if all functions would have JSDoc
lack of returns would mean that this function does not return anything

File: front/scripts/baseline/mercury/utils/state.ts [hunk 3]
```diff
@@ -10,6 +10,13 @@ interface ObjectProperties {
 }
 
 module Mercury.Utils {
+	/**
+	 * @param {string} [str='']
```
Comment (hakubo) [inline front/scripts/baseline/mercury/utils/state.ts]: I don't see any default for str in function body

File: front/scripts/baseline/mercury/utils/state.ts [hunk 4]
```diff
@@ -10,6 +10,13 @@ interface ObjectProperties {
...
+	 * @param {string|object} [ns=window]
+	 * @param {any} val
```
Comment (hakubo) [inline front/scripts/baseline/mercury/utils/state.ts]: I think in JSDoc this should be {?object} which means any object or null

File: front/scripts/main/routes/CuratedContentEditorBlockAddItemRoute.ts [hunk 5]
```diff
@@ -4,10 +4,19 @@
 'use strict';
 
 App.CuratedContentEditorBlockAddItemRoute = Em.Route.extend({
+	/**
+	 * @returns {CuratedContentEditorItemModel}
```
Comment (hakubo) [inline front/scripts/main/routes/CuratedContentEditorBlockAddItemRoute.ts]: Can't find @typedef of it

File: front/scripts/main/routes/CuratedContentEditorRoute.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ App.CuratedContentEditorRoute = Em.Route.extend(
 		/**
 		 * @returns {void}
 		 */
-		beforeModel() {
+			beforeModel() {
```
Comment (rogatty) [inline front/scripts/main/routes/CuratedContentEditorRoute.js]: Indentation is broken in this file.

File: front/scripts/auth/common/AuthLogger.ts [hunk 7]
```diff
@@ -1,24 +1,36 @@
...
-	status: number;
-	response: any;
-	heliosUrl: string;
-	clientUrl: string;
-}
-
+/**
+ * ClickStreamPayload
+ * @typedef {object} ClickStreamPayload
+ * @property {Object[]} events
```
Comment (bognix) [inline front/scripts/auth/common/AuthLogger.ts]: capital Object

File: front/scripts/baseline/mercury/utils/isPrimitive.ts [hunk 8]
```diff
@@ -1,4 +1,8 @@
 module Mercury.Utils {
+	/**
+	 * @param {any} val
```
Comment (bognix) [inline front/scripts/baseline/mercury/utils/isPrimitive.ts]: replace any with *

File: front/scripts/baseline/mercury/utils/loadScript.ts (hunk 2 above)
Comment (bognix) [inline front/scripts/baseline/mercury/utils/loadScript.ts]: the problem is that eslint can either force you to add jsdocs everywhere or nowhere.
You can't do something like: don't force jsdocs when there is no return value.

But don't get me wrong, I totally agree with you, it introduces unnecessary noise

File: front/scripts/baseline/mercury/utils/loadScript.ts [hunk 9]
```diff
@@ -2,11 +2,21 @@
...
+	 */
 	function loadError (error: ErrorEvent): void {
 		var scriptTag: HTMLScriptElement = <HTMLScriptElement> error.target;
 		throw new URIError('The script ' + scriptTag.src + ' is not accessible.');
 	}
 
+	/**
+	 * @param {string} src
+	 * @param {Function} [fOnload]
```
Comment (bognix) [inline front/scripts/baseline/mercury/utils/loadScript.ts]: fOnload does't look like optional parameter

File: front/scripts/baseline/mercury/utils/state.ts [hunk 10]
```diff
@@ -10,6 +10,13 @@ interface ObjectProperties {
 }
 
 module Mercury.Utils {
+	/**
+	 * @param {string} [str='']
+	 * @param {string|object} [ns=window]
```
Comment (bognix) [inline front/scripts/baseline/mercury/utils/state.ts]: ns does't look like it optional parameter

File: front/scripts/main/CurrentUser.ts [hunk 11]
```diff
@@ -56,6 +63,10 @@ App.CurrentUser = Em.Object.extend({
 		M.prop('userLanguage', userLanguage);
 	},
 
+	/**
+	 * @param {QueryUserInfoResponse} result
```
Comment (bognix) [inline front/scripts/main/CurrentUser.ts]: This type is not defined anywhere.
Are you sure JSDoc understands it?

File: front/scripts/main/components/ArticleContentComponent.ts [hunk 12]
```diff
@@ -5,289 +5,370 @@
...
-			    $contributionComponent: JQuery = null;
-
-			headers.forEach((header: ArticleSectionHeader): void => {
-				$contributionComponent = this.createArticleContributionComponent(header.section);
-				$sectionHeader = this.$(header.element);
-				$sectionHeader.prepend($contributionComponent).addClass('short-header');
-				$contributionComponent.wrap('<div class="icon-wrapper"></div>');
+		/**
+		 * Generates table of contents data based on h2 elements in the article
+		 * TODO: Temporary solution for generating Table of Contents
```
Comment (bognix) [inline front/scripts/main/components/ArticleContentComponent.ts]: use @TODO - it's official JSDoc notation

File: front/scripts/main/components/ArticleContentComponent.ts [hunk 13]
```diff
@@ -5,289 +5,370 @@
...
+					height: parseInt(element.getAttribute('height'), 10),
+					imgWidth: element.offsetWidth,
+					media: media,
+				}).createElement();
+
+			return component.$().attr('data-ref', ref);
+		},
+
+		/**
+		 * @param {App.ArticleModel} model
```
Comment (bognix) [inline front/scripts/main/components/ArticleContentComponent.ts]: is ArticleModel defined anywhere - in JSDocs?

**Comment by tomasznapieralski** [comment]
front and server passed locally for me

**Pull request merged**